nailgun/entities.py

```diff
@@ -2031,6 +2031,17 @@ def __init__(self, server_config=None, **kwargs):
         }
         super(ContentCredential, self).__init__(server_config, **kwargs)
 
+    def update_payload(self, fields=None):
```

Comment:
I think this behavior would be great to implement within `EntityUpdateMixin.update_payload` directly, so that the behavior is available to all entities using this mixin.

Each entity could define a class attribute of update-able fields, with a default behavior (no attribute defined) of all fields being update-able. 

This way, you wouldn't have to add this to every entity, but for any entity where you want to limit the fields included in update_payload, you just define `Entity.updateable_fields` as a list of field names.

The `EntityUpdateMixin.update_payload` method can then filter its return on `self.updateable_fields`.

Reply:
Yes, I could do that. I wasn't sure if we still want to do such a big change.
Anyway, that would require quite a lot of manual work and testing. How about splitting it to three PRs, where one would implement the functionality for the mixin (with the default of updatable fields == all fields, which is the current state), another would specify the fields for the entities I need here and the last one would eventually specify them for all the remaining entities?

Reply:
I don't think its necessary to specify updateable fields for every entity - the `EntityUpdateMixin.update_payload` could safely reference `self.updateable_fields` without it being defined on all entities. For the entities where it is defined, it would be used. For those without it defined, the current behavior would remain consistent. 

As far as testing goes, the existing unit test structure could be updated to cover the new behavior.  You wouldn't need to test it with every "real" entity in nailgun.
https://github.com/SatelliteQE/nailgun/blob/master/tests/test_entity_mixins.py#L850

I'd prefer to scope it into a single PR, as defining updateable fields on the three entities included in this PR would make the updates to EntityUpdateMixin actually useful.